### PR TITLE
Api 5277 update oauth proxy depdencies

### PR DESCRIPTION
**Pull request by lukemarkwordtlibertyits**
# Description

Update all outdated dependencies in the oauth-proxy.

**Changes**

- [x] @okta/okta-sdk-nodejs 4.2.0 -> 4.4.0
- [x] aws-sdk 2.792.0 -> 2.859.0
- [x] cls-rtracer 2.5.1 -> 2.6.0
- [x] eslint 7.13.0 -> 7.21.0
- [x] eslint-plugin-jest 24.1.3 -> 24.1.9
- [x] eslint-plugin-prettier 3.1.4 -> 3.3.1
- [x] validator 13.1.17 -> 13.5.2
- [x] @sentry/node 5.27.4 -> 5.30.0

**Dependencies that will need special attention in the future**
|Package                | Current  | Wanted  | Latest|
|---|---|---|--- |
|mock-express-response |    0.2.2  |  0.2.2  |  0.3.0 |
|prettier               |   2.0.5  |  2.0.5  |  2.2.1 |
|eslint-config-prettier  | 6.15.0 |  6.15.0  |  8.1.0  |
|express-prom-bundle     |  5.1.5  |  5.1.5  |  6.3.4 |
|jwt-decode              |  2.2.0  |  2.2.0  |  3.1.2 |
|openid-client          | 3.15.10 | 3.15.10  |  4.4.2 |
|prom-client            |  11.5.3  | 11.5.3  | 13.1.0  |
|yargs                  |  15.4.1 |  15.4.1  | 16.2.0 |
|@sentry/node |             5.27.4  | 5.30.0   | 6.2.1 |

# AC

- [x] `npm outdated` should only return dependencies whose wanted version is different then the latest (see table above)
  - [x] a story for these dependencies should be created 
  https://vajira.max.gov/browse/API-5630
- [x] All unit tests should pass
- [x] All regression tests should pass

### CODE REVIEW

**Comment by lukemarkwordtlibertyits** [comment]
# Local Auth Testing

## Confirm Happy Path is working.

- Make Authorization request.
- client_store = local
- client_store has redirect `localhost:8080/redirect`

### Expected results

- [ ] Redirect to authorization server 

## Confirm Client Redirect Miss is working.

- Make Authorization request.
- client_store = local
- client_store does not have redirect `localhost:8080/redirect`

expected results

- [ ] expected json
```json
{
    "error":	"invalid_client",
    "error_description": "The redirect URI specified by the application does not match any of the registered redirect URIs. Erroneous redirect URI: http://localhost:8080/redirect"
}
```

## Client miss in client_store

- Make Authorization request.
- client_store = local
- specified client not in client_store

expected results

- [ ] expected json
```json
{
    "error": "invalid_client",
    "error_description": "Invalid client for the authorization path",
}
```

## No Clients in client_store

- Make Authorization request.
- client_store = local
- force dynamo error 

expected results

- [ ] expected json
```json
{
    "error": "invalid_client",
    "error_description": "The client specified by the application is not valid."
}
```

## No Okta Redirect URI

- Make Authorization request.
- client_store = null
- redirect uri not registered with okta

expected results

- [ ] expected json
```json
{
    "error": "invalid_client",
    "error_description": "The redirect URI specified by the application does not match any of the registered redirect URIs. Erroneous redirect URI: ${client_redirect}"
}
```

## No Okta Client Id

- Make Authorization request.
- client_store = null
- client id not registered with okta

expected results

- [ ] expected json
```json
{
    "error": "invalid_client",
    "error_description": "The client specified by the application is not valid.",
}
```

**Comment by beaugrantham-va** [comment]
I spot-checked a random dependency that wasn't listed in the changes or the future changes and there appears to be a newer minor version.

```
"express": "^4.16.4"
```

The latest express claims to be 4.17.1 - https://www.npmjs.com/package/express

File: oauth-proxy/package.json [hunk 1]
```diff
@@ -46,14 +46,14 @@
...
+    "@okta/okta-sdk-nodejs": "^4.4.0",
+    "@sentry/node": "^5.30.0",
+    "aws-sdk": "^2.859.0",
     "axios": "^0.21.1",
     "body-parser": "^1.18.3",
-    "cls-rtracer": "^2.5.1",
+    "cls-rtracer": "^2.6.0",
     "cors": "^2.8.5",
-    "eslint-plugin-jest": "^24.1.0",
+    "eslint-plugin-jest": "^24.1.9",
```
Comment (beaugrantham-va) [inline oauth-proxy/package.json:56]: Unrelated to the version change, but should this be in `devDependencies` instead of `dependencies`?

**Review by crolarlibertyva — DISMISSED** [review]

**Review by beaugrantham-va — DISMISSED** [review]

File: oauth-proxy/tests/scripts/token_tests.sh [hunk 2]
```diff
@@ -92,22 +92,22 @@ do_client_credentials() {
   local network=""
   if [[ $HOST == *"dev"* ]];
   then
-    url="https://dev-api.va.gov/oauth2/health-insurance/v1"
+    url="https://dev-api.va.gov/oauth2/health/system/v1"
```
Comment (crolarlibertyva) [inline oauth-proxy/tests/scripts/token_tests.sh:95]: Can we update the dev-config-base.json to go with the changes in this test?

**Review by crolarlibertyva — CHANGES_REQUESTED** [review]

File: oauth-proxy/tests/scripts/token_tests.sh (hunk 2 above)
Comment (beaugrantham-va) [reply oauth-proxy/tests/scripts/token_tests.sh:95]: The dev-config.base.json already includes the new `/health/system` route. The old route is being retained until we fully decommission production/sandbox consumers to keep in sync.

**Review by beaugrantham-va — APPROVED** [review]

**Review by crolarlibertyva — APPROVED** [review]

**Pull request merged**